randutil/rand.go

```diff
@@ -70,26 +110,29 @@ const letters = "BCDFGHJKLMNPQRSTVWXYbcdfghjklmnpqrstvwxy0123456789"
 // chance. Numbers are included.
 //
 // Not cryptographically secure.
-func RandomString(length int) string {
+func (r *PseudoRand) RandomString(length int) string {
```

Comment:
Given this is in randutil on the `PseudoRand` type, perhaps we could just call this method `String` and the one below `Duration`? (The signature of `String(length int) string` won't clash with `fmt.Stringer` either.)

Reply:
This API has been in use for quite some time in both Pebble and Snapd, and the change will generate a significant number of changes in the snapd codebase. But super willing to make this as nice as possible.

```
somePrngInst := randutil.NewPseudoRand(nil)
somePrngInst.String(12)
somePrngInst.Duration(12)
```

@niemeyer do you have a preference here? 

```
randutil.RandomString(12): somePrngInst.String(12) or somePrngInst.RandomString(12)
randutil.RandomDuration(12): somePrngInst.Duration(12) or somePrngInst.RandomDuration(12)
```

Reply:
I'm not sure I understand the context for the proposal.  It feels a bit over engineered, which I guess is also Ben's point, but you're saying that this is already in use in snapd, which needs care indeed. At the same time, it looks like most of these functions and the type are new, so what's actually in use in snapd that requires this functionality?  Why isn't it just RandomString and RandomDuration still?  I would prefer to not be forced to use an API that requires us to pass along a type to all corners of the code that could possibly want to use a random string or value of whatever kind.

Reply:
Hi Gustavo. At this stage only a proposed change for snapd exist here (https://github.com/snapcore/snapd/pull/12759), with the goal of first having an agreement on this side.

I was trying to make a proposal for randutil which would make it more flexible as a x-go library, primarily enabling the consumer to select or provide a seed (my interpretation of your comment [here](https://github.com/canonical/x-go/pull/17)). Since ```RandomString``` and ```RandomDuration```, even in snapd was used in different contexts, their seeding requirements were different. Learning from this, I thought it would be better to enable the user to create a pseudo random instance with a required seed, instead of the previous global seed / prng shared by both.

Perhaps this is all over engineering in hindsight.

Reply:
OK agreed - we can quickly chat about this in our next call. Thanks.